### PR TITLE
fix(issuer): raise the metadata signing limit, and the transport limit under it

### PR DESCRIPTION
**Pull request by leifj**
## The problem

An issuer offering nine credential types, three of them with registry-sourced
artwork, publishes **2.19 MiB** of Credential Issuer Metadata: each
`credential_configurations_supported` entry now inlines that type's display
metadata, logos included. `SignMetadata`'s 2 MiB guard rejects it:

```
ERROR apigw.apiv1  signed_metadata unavailable, serving unsigned metadata
  {"error": "failed to sign metadata via issuer: rpc error: code = InvalidArgument
   desc = metadata_json is too large"}
```

APIGW treats that as non-fatal and serves the metadata unsigned, so nobody
sees an error. It is a silent downgrade for every wallet that would otherwise
have verified `signed_metadata`.

Sizes from the deployment where this was found, for scale:

| type | bytes |
|---|---|
| ehic | 1,340,851 |
| pid_1_8 | 357,067 |
| pid_1_5 | 356,263 |
| whole document | 2,188,352 |

## The change

Raising the guard on its own would not have been enough: the payload crosses
gRPC, and both ends still used grpc-go's 4 MiB default, so the failure would
have moved from a specific `InvalidArgument` to a generic transport error a
little further along. This therefore sets one limit for both ends and keeps
the application guard below it:

- `grpchelpers.MaxMessageBytes` (16 MiB) applied as `MaxRecvMsgSize` /
  `MaxSendMsgSize` on the server and as default call options on the client,
  so a call is not silently bounded by whichever end was left at the default.
- `maxMetadataJSONBytes` 2 MiB → 8 MiB, preserving what its comment already
  promised: this check, not the transport's, is the one that fires, with a
  clear error.

`NewServerOptions` previously returned `nil` when TLS was disabled; it now
returns the message-size options in that case too, since the limit is about
payload size rather than transport security.

## Tests

The existing boundary cases are written in terms of the constant, so they
continue to hold at the new value. Added a test pinning the two properties
the value depends on: the guard must stay below the transport limit, and the
size actually observed in the field must be signable. The server-option count
assertions now account for the two size options.

`go test ./internal/issuer/apiv1/ ./pkg/grpchelpers/` passes.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_01QLweCnieT6BLQiz8vecKji

### CODE REVIEW

File: pkg/grpchelpers/server.go [hunk 1]
```diff
@@ -25,9 +25,28 @@ import (
...
+//
+// Application-level guards stay below this deliberately, so an oversized
+// payload gets a specific error from the service rather than a generic
+// transport failure - see the issuer's maxMetadataJSONBytes.
+const MaxMessageBytes = 16 * 1024 * 1024
+
 func NewServerOptions(cfg model.GRPCServer) ([]grpc.ServerOption, error) {
+	sizeOpts := []grpc.ServerOption{
+		grpc.MaxRecvMsgSize(MaxMessageBytes),
+		grpc.MaxSendMsgSize(MaxMessageBytes),
```
[the remark this answers is not in the document]
Comment (leifj) [reply pkg/grpchelpers/server.go:46]: Added in bf48761e: a round trip over the real constructors, with a 6 MiB payload (above grpc-go's 4 MiB default, below MaxMessageBytes) asserted to cross intact in both directions, plus one past MaxMessageBytes asserted to fail with ResourceExhausted so the limit is shown to be a real bound.

File: pkg/grpchelpers/server.go [hunk 2]
```diff
@@ -25,9 +25,28 @@ import (
...
+// carries one entry per configured scope, and each entry now inlines that
+// type's display metadata - logos included - so an issuer offering a handful
+// of types can publish several MiB. It is signed by the issuer over gRPC
+// (SignMetadata), so the transport limit is what decides whether an issuer
+// can sign its own metadata at all.
+//
+// Application-level guards stay below this deliberately, so an oversized
+// payload gets a specific error from the service rather than a generic
+// transport failure - see the issuer's maxMetadataJSONBytes.
+const MaxMessageBytes = 16 * 1024 * 1024
```
[the remark this answers is not in the document]
Comment (leifj) [reply pkg/grpchelpers/server.go:36]: Fixed in bf48761e. The constant now carries its own comment and NewServerOptions' doc is restored, with the stale "returns nil" line replaced by a note that the message-size options are always included.
